src/indexer/stamper.rs

```diff
@@ -92,7 +32,8 @@ impl Stamper {
 
     /// Reverts the stamper to a given `Opstamp` value and returns it
     pub fn revert(&self, to_opstamp: Opstamp) -> Opstamp {
-        self.0.revert(to_opstamp, Ordering::SeqCst)
+        self.0.store(to_opstamp, Ordering::SeqCst);
```

Comment:
by the way I think Relaxed ordering was ok here.
```suggestion
        self.0.store(to_opstamp, Ordering::SeqCst);
```

Reply:
Get it

Reply:
ahh, ok